### PR TITLE
DOC/INSTALL: Update OpenMPI build instructions

### LINKED ISSUE #1966: How to install openmpi with ucx?
According to INSTALL file:

> Compiling OpenMPI with UCX library
> ==================================
> 
> The Open MPI package includes two public software layers: MPI and OpenSHMEM.
> UCX support in OpenMPI is not upstream yet, it is work in progress.
> Current work on OpenMPI/UCX integration is ongoing for OpenSHMEM API and can be found here:
> 
> https://github.com/openucx/ompi-mirror/pull/1
> 
> 
> To compile OpenMPI with UCX support:
> 
> # checkout Pull Request
> % git clone https://github.com/openucx/ompi-mirror
> % cd ompi-mirror
> % git fetch origin pull/1/head:pr-1
> % git checkout pr-1

That's repository does not exist!

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #1966

### CODE REVIEW

**Comment by shamisp** [comment]
@tonycurtis please take a look.
@yosefe general question - shell we keep it as a separate file or move it to readme ?

**Comment by yosefe** [comment]
i'd say move to README. Better have all docs in one place.

**Comment by shamisp** [comment]
@yosefe Is there any "distro" or GNU guidance for this ?

**Comment by tonycurtis** [comment]
> On Mar 6, 2019, at 9:46 AM, Pavel Shamis (Pasha) <notifications@github.com> wrote:
> 
> @tonycurtis <https://github.com/tonycurtis> please take a look.
> 

added some comments in PR

how about putting things like this somewhere under the doc directory?

tony

**Comment by shamisp** [comment]
IMHO readme is better. I would follow distro guidance is there is any.

**Comment by miked-mellanox** [comment]
maybe best to have docs/openmpi.txt with instructions. ucx is used not only from OpenMPI, also mpich and others (with bindings). INSTALL should contain info on ucx specific install procedures. WDYT

**Comment by tonycurtis** [comment]
> On Mar 6, 2019, at 10:25 AM, Mike Dubman <notifications@github.com> wrote:
> 
> maybe best to have docs/openmpi.txt with instructions. ucx is used not only from OpenMPI, also mpich and others (with bindings). INSTALL should contain info on ucx specific install procedures. WDYT
> 

I like to think of READMEs as launchpads: short & general meta-info so people don’t zone out, and then pointers to more specific info.

tony

**Comment by miked-mellanox** [comment]
> On Mar 6, 2019, at 10:25 AM, Mike Dubman ***@***.***> wrote: maybe best to have docs/openmpi.txt with instructions. ucx is used not only from OpenMPI, also mpich and others (with bindings). INSTALL should contain info on ucx specific install procedures. WDYT
> I like to think of READMEs as launchpads: short & general meta-info so people don’t zone out, and then pointers to more specific info. tony

generally agree. but ompi build instructions are out of scope of INSTALL file or README file of UCX. 
what about ucx java/py bindings apps? mpich? else?
README file can contain references to docs/___ for high level apps instructions.

**Comment by shamisp** [comment]
@miked-mellanox I thing you are on the same page with @tonycurtis . @tonycurtis suggests to reference/link the instructions from readme.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9167/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9503/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9504/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9505/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp @tonycurtis can you pls take a look?

**Comment by yosefe** [comment]
@tonycurtis i've copied it from @shamisp's test in INSTALL, probably better to remove it

**Comment by shamisp** [comment]
@yosefe Personally I'm okay within having this as a part of the readme. This is one of the first things that show ups when you load GitHub page and it is the first question that users would ask.

**Comment by yosefe** [comment]
@shamisp are you ok with removing it? i also think we should remove it,

**Comment by shamisp** [comment]
Remove and put where? We cannot just remove it without any alternatives.

**Comment by yosefe** [comment]
IMHO we don’t need to explain how makefile works. Just how to use it in UCX.
Let me rephrase the text and upload new patch.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9672/ for details (Mellanox internal link).
